tests/CMakeLists.txt

```diff
@@ -5,24 +5,12 @@ set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_FULL_LIBDIR})
 add_library(quda_test STATIC googletest/src/gtest-all.cc)
 target_include_directories(quda_test SYSTEM PUBLIC  googletest/include googletest)
 target_include_directories(quda_test SYSTEM PUBLIC ${CUDAToolkit_INCLUDE_DIRS})
-target_include_directories(quda_test SYSTEM PUBLIC ${EIGEN_INCLUDE_DIRS})
+target_link_libraries(quda_test PUBLIC Eigen)
 target_link_libraries(quda_test PUBLIC quda)
-target_compile_options(quda_test PUBLIC -fPIC)
+set_target_properties(quda_test PROPERTIES POSITION_INDEPENDENT_CODE ON)
 add_subdirectory(utils)
 add_subdirectory(host_reference)
 
-if(QUDA_QIO
-   AND QUDA_DOWNLOAD_USQCD
-   AND NOT QIO_FOUND)
-  add_dependencies(quda_test QIO)
-endif()
-
-if(QUDA_QMP
-   AND QUDA_DOWNLOAD_USQCD
-   AND NOT QMP_FOUND)
-  add_dependencies(quda_test QMP)
-endif()
-
 if(QUDA_NVSHMEM AND QUDA_DOWNLOAD_NVSHMEM)
   add_dependencies(quda_test NVSHMEM)
```

Comment:
Should NVSHMEM be a dependency of the quda library to make it self contained?

Reply:
Until we build nvshmem with cmake itself this is a hacky workaround we still need (I believe).